### DISCUSSION TITLE
No device discovered, what I am doing wrong ??

### DISCUSSION THREAD

**Discussion by TheJokerHA** [leikoilja/ha-google-home#478] [General]
Hi all,

no device discovered, what I am doing wrong ?? I wolud like to cotrol my gate via HA.
![Google home](https://user-images.githubusercontent.com/85641621/156724173-d31a1b9f-afa6-49b6-8c0e-ff70fcf32946.JPG)
![Screenshot_20220304-090839_Home](https://user-images.githubusercontent.com/85641621/156724581-47584092-a754-4cca-881c-876cbe687f25.jpg)

HA is running on my QNAP server as virtualization is there maybe something there what is generating the issue ??

BR
Nedzad

**Comment by leikoilja** [comment]
Could be that your HomeAssistance cannot auto-discover devices on your network. If that's the case, you would need to take a closer look at your network connections 🤔 Can you also check your logs? This integration is only discovering devices that are Google Home/Assistant compatible, so your garage sensor/device will not be discovered. You should be able to see that from the logs

**Reply by TheJokerHA** [reply]
In the log file from HA I can´t find anything about it. Where shut I look are there special log files regarding this application??

**Reply by leikoilja** [reply]
Please take a look at REAME dedicated section here - https://github.com/leikoilja/ha-google-home#troubleshooting
Once you obtain more logs and discover what could be the problem - feel free to open a new issue with all information included

**Comment by ustix** [comment]
Hi there @leikoilja!

I have successfully installed your integration with a twist: get_tokens/glocaltokens were not working to get master token, so I have used the windows way to get it. That was successful so I have used that master token to login to home assistant and it worked, but now I don't see any devices automatically attached, but I have more than twenty devices (lights, switches, temperature sensors, AC-s, cameras, vacuum cleaner) in Google Home so I'm not sure what to expect from this integration. My home assistant device is a Radxa Zero (4GB) with Armbian 23.8 Bookworm (6.1.15 kernel), I have urllib3 version 1.26 and gpsoauth python3.11 installed. Also, as above I have zero information about errors in my log. Do you have any idea?

**Reply by leikoilja** [reply]
@ustix, this integration is easy to misunderstand due to bad Google product namings. And i dont blame you for that.
but basically google has google home products like these: 
![image](https://github.com/leikoilja/ha-google-home/assets/10655107/71b09971-79e8-4405-8ccc-6a7acb749841)
and on top of those you can add various sensors to your google home eco-system.
So this integration only enables you to add google home speakers into home assistant and control them using home assistant. stuff like timers/alarms/do not disturb mode.
all other sensors, lights and switches are out of the score of this integration. you will most likely want to check [this](https://www.home-assistant.io/integrations/google_assistant/) to be able to add them to your home assistant instance.

what i suggest you do to is to remove them from your google ecosystem and migrate them to home assistant directly using their supported integrations to have much better support and control over them.
for example, if you have philips hue lights, instead of adding them to google home ecosystem and then trying to expose them in home-assistant, go direct route and add them using hue [dedicated integration](https://www.home-assistant.io/integrations/hue/).

**Reply by ustix** [reply]
I'm really grateful for your answer. I've come to a conclusion before you wrote that, that this is the situation here, and I won't get any device from Google Home, but I wasn't sure. I have an Alexa Eco-system, but I am also integrated all my devices to Google Home, because that is much nicer in automation and UI (Alexa won't even load the UI sometimes), but these Google devices are so ugly that I won't buy them for any price.
I wanted to integrate devices from Google cloud because some of my devices (for example my AC - Hisense with ConnectLife app) doesn't have Home Assistant integration yet, and also cloud integration is easier than adding the devices one by one.
Thank you for your answer again!

**Reply by leikoilja** [reply]
no probs ;) 
i have been through the same struggle. moved to Alexa speakers for automations (since in my opinion they are better than google speakers for customer automations) and had to move all my sensors/switches to zigbee to properly connect them 'locally' to home assistant. i am happy now with a setup and choose my devices based on the support they have for zigbee (and home assistant) haha 😅 
something you can try for your AC is broadlink device for "replicating" your AC remote by sending IR commands. I have used it for my old school projector and it worked pretty well. got it straight from aliexpress for not that much. Broadlink has a decent integration with home assistant and i was able to setup some automations that i wanted

**Reply by ustix** [reply]
The AC is not that much of a struggle, I can use it in Google or Alexa, or in it's own app, but mostly it is just turned on or off, and it will be automatized for the temperatures in the house, and I think the integration is in progress (at least the public api for it).[https://community.home-assistant.io/t/add-integration-for-hisense-devices-using-connectlife/459082/16]
So I am not disappointed.
Also I haven't found a way in Alexa automation to use triggers like "when a device state is ON" or at least for the devices I need (like lights). It has that only for "smart home" devices (the house icon ones with the wifi symbol in it), which is strange for me, cause in Google there is no such difference, and I think this is a very basic thing. I want my switches for lights to be synced with my bulbs state. So I've just moved to Home Assistant, where almost everything is possible (if it has integration then really everything :D)
Thanks again for your fast reply!
